actions/update-constraints/generate_constraints.sh

```diff
@@ -7,14 +7,13 @@ if [ "$(uname)" != "Linux" ]; then
     exit 1
 fi
 
-# If we run from tox, ignore the index url
-unset PIP_EXTRA_INDEX_URL
-
 # Read list of requirements files from the config file
 REQUIREMENTS_FILES=$(grep -v '^#' requirements-files.in | tr '\n' ' ')
 
 CONSTRAINTS_FILE=constraints-dev.txt
 
+export UV_INDEX_STRATEGY=unsafe-best-match
```

Comment:
it's called unsafe because there's an attack vector: https://docs.astral.sh/uv/concepts/indexes/#searching-across-multiple-indexes but it applies to any pip runs. Since we use pip anyway, it doesn't seem to make it any worse than status quo.